### PR TITLE
P2: LILA-5572/update-redeemauto-function-to-work-like-autoredeem2-in-sdk

### PR DESCRIPTION
**Pull request by GigaHierz**
- mark `autoRedeem2` as deprecated

### CODE REVIEW

File: src/typechain/IToucanPoolToken.d.ts [hunk 1]
```diff
@@ -382,7 +382,12 @@ export class IToucanPoolToken extends BaseContract {
 
     getScoredTCO2s(overrides?: CallOverrides): Promise<string[]>;
 
-    redeemAuto(amount: BigNumberish, overrides?: CallOverrides): Promise<void>;
+    redeemAuto(
+      amount: BigNumberish,
+      overrides?: CallOverrides
+    ): Promise<
+      [string[], BigNumber[]] & { tco2s: string[]; amounts: BigNumber[] }
```
Comment (Niwilai) [inline src/typechain/IToucanPoolToken.d.ts:389]: Is this generated?

File: src/index.ts [hunk 2]
```diff
@@ -311,7 +311,7 @@ export default class ToucanClient {
   redeemAuto = async (
     pool: PoolSymbol,
     amount: BigNumber
-  ): Promise<ContractReceipt> => {
+  ): Promise<{ address: string; amount: BigNumber }[]> => {
```
Comment (Niwilai) [inline src/index.ts:314]: Can we give this a `RedeemAutoResponse` type or similar?

**Review by Niwilai — COMMENTED** [review]
Good job, looking good. Some minor comments:

File: src/typechain/IToucanPoolToken.d.ts (hunk 1 above)
Comment (GigaHierz) [reply src/typechain/IToucanPoolToken.d.ts:389]: yes

**Review by Niwilai — APPROVED** [review]

**Pull request merged**